python/ray/autoscaler/gcp/node_provider.py

```diff
@@ -158,7 +189,7 @@ def create_node(self, base_config, tags, count):
 
         name_label = labels[TAG_RAY_NODE_NAME]
         assert (len(name_label) <=
-                (INSTANCE_NAME_MAX_LEN - INSTANCE_NAME_UUID_LEN - 1)), (
+                (INSTANCE_NAME_MAX_LEN - INSTANCE_NAME_UUID_LEN - 3)), (
```

Comment:
I decreased this by 2 since the name length is compared against head name label, whereas the same limit applies to worker name label, which is 2 characters longer.